.github/workflows/main-workflow.yml

```diff
@@ -0,0 +1,30 @@
+name: Pipe
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8
+          pip install coverage
+          pip install pyinstaller
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          # exit-zero treats all errors as warnings.
+          flake8 . --count --exit-zero --max-line-length=80 --ignore=E111,E121,E126 --statistics
+      - name: build
+        run: g++ -Iincludes -DUNIT_TEST -Wall -Wextra -fprofile-arcs -ftest-coverage -o testchecker ./test/*.c ./src/bms-sender.c ./src/bms-exec.c
+      - name: Pipe run
+        run: ./testchecker | python receiverTest.py
```

Comment:
receiverTest does not take any input from the console, so the console output of testchecker will be ignored